lib-user tests fail in a non-UTC time zone

## Package
- lib-user

## Describe the bug
`lib-user` date-time tests consistently fail on my laptop (currently UTC+1) but pass in CI.

```sh
163 passing (2s)
  1 pending
  2 failing

  1) components > MainContent > getDateRangeSelectOptions
       with a custom date range
         should return the expected date range options:

      AssertionError: expected [ Array(8) ] to deeply equal [ Array(8) ]
      + expected - actual

       [
         {
           "label": "LAST 7 DAYS"
      -    "value": "2023-04-08"
      +    "value": "2023-04-09"
         }
         {
           "label": "LAST 30 DAYS"
      -    "value": "2023-03-16"
      +    "value": "2023-03-17"
         }
         {
           "label": "THIS MONTH"
           "value": "2023-04-01"
         }
         {
           "label": "LAST 3 MONTHS"
      -    "value": "2023-01-14"
      +    "value": "2023-01-15"
         }
         {
           "label": "THIS YEAR"
           "value": "2023-01-01"
      
      at Context.<anonymous> (src/components/shared/MainContent/helpers/getDateRangeSelectOptions.spec.js:172:40)
      at processImmediate (node:internal/timers:478:21)

  2) components > MainContent > getDateRangeSelectOptions
       with a source created_at date
         should return an All Time option with the source created_at date:

      AssertionError: expected [ Array(8) ] to deeply equal [ Array(8) ]
      + expected - actual

       [
         {
           "label": "LAST 7 DAYS"
      -    "value": "2023-04-08"
      +    "value": "2023-04-09"
         }
         {
           "label": "LAST 30 DAYS"
      -    "value": "2023-03-16"
      +    "value": "2023-03-17"
         }
         {
           "label": "THIS MONTH"
           "value": "2023-04-01"
         }
         {
           "label": "LAST 3 MONTHS"
      -    "value": "2023-01-14"
      +    "value": "2023-01-15"
         }
         {
           "label": "THIS YEAR"
           "value": "2023-01-01"
--
           "label": "ALL TIME"
           "value": "2020-04-15"
         }
         {
      -    "label": "APR 9 - 15"
      +    "label": "CUSTOM"
           "value": "custom"
         }
       ]
      
      at Context.<anonymous> (src/components/shared/MainContent/helpers/getDateRangeSelectOptions.spec.js:241:40)
      at processImmediate (node:internal/timers:478:21)



error Command failed with exit code 2.
```

## To Reproduce
I can reproduce this consistently on my laptop, to the point that I just ignore failing tests. I'm not sure how to reproduce it elsewhere. Maybe change the time zone on the machine that runs the tests?

## Expected behavior
Tests should pass or fail consistently in both CI and locally.

--- a/packages/lib-user/src/components/shared/MainContent/helpers/getDateRangeSelectOptions.spec.js
+++ b/packages/lib-user/src/components/shared/MainContent/helpers/getDateRangeSelectOptions.spec.js
@@ -154,7 +154,7 @@ describe('components > MainContent > getDateRangeSelectOptions', function () {
   describe('with a custom date range', function () {
 
     beforeEach(function () {
-      clock = sinon.useFakeTimers(new Date('2023-04-15T00:00:00'))
+      clock = sinon.useFakeTimers(new Date('2023-04-15T00:00:00Z'))
     })
 
     afterEach(function () {
@@ -223,7 +223,7 @@ describe('components > MainContent > getDateRangeSelectOptions', function () {
   describe('with a source created_at date', function () {
 
     beforeEach(function () {
-      clock = sinon.useFakeTimers(new Date('2023-04-15T00:00:00'))
+      clock = sinon.useFakeTimers(new Date('2023-04-15T00:00:00Z'))
     })
 
     afterEach(function () {

